deepchecks/nlp/utils/text_embeddings.py

```diff
@@ -9,14 +9,52 @@
 # ----------------------------------------------------------------------------
 #
 """Utils module for calculating embeddings for text."""
+import sys
+from itertools import islice
 from typing import Optional
 
 import numpy as np
 from tqdm import tqdm
 
+EMBEDDING_MODEL = 'text-embedding-ada-002'
+EMBEDDING_CTX_LENGTH = 8191
+EMBEDDING_ENCODING = 'cl100k_base'
+
+
+def batched(iterable, n):
+    """Batch data into tuples of length n. The last batch may be shorter."""
+    # batched('ABCDEFG', 3) --> ABC DEF G
+    if n < 1:
+        raise ValueError('n must be at least one')
+    it = iter(iterable)
+    while True:
+        batch = tuple(islice(it, n))
+        if not batch:
+            return
+        yield batch
+
+
+def encode_text(text, encoding_name):
+    """Encode tokens with the given encoding."""
+    # tiktoken.get_encoding is only available in python 3.8 and above.
+    # This means that for python < 3.8, the batching is just using chunk_length chars each time.
+    if sys.version_info >= (3, 8):
+        import tiktoken  # pylint: disable=import-outside-toplevel
+        encoding = tiktoken.get_encoding(encoding_name)
+        return encoding.encode(text)
+    else:
+        return text
+
+
+def iterate_batched(tokenized_text, chunk_length):
+    """Chunk text into tokens of length chunk_length."""
+    chunks_iterator = batched(tokenized_text, chunk_length)
+    yield from chunks_iterator
+
 
 def calculate_builtin_embeddings(text: np.array, model: str = 'miniLM',
-                                 file_path: Optional[str] = 'embeddings.npy') -> np.array:
+                                 file_path: Optional[str] = 'embeddings.npy',
+                                 device: Optional[str] = None) -> np.array:
```

Comment:
we discussed an argument to control what to do with texts longer then limit and that by default will be average with warning/exception

Reply:
the warnings / exception should state with samples were longer then limit + what is the limit (which is based on selected model i guess)

Reply:
Can't state which samples, can be too many. Will length of maximal one